src/state/dataSelectors.ts

```diff
@@ -16,9 +18,25 @@ export const selectChartDataWithIndexes = (state: RechartsRootState): ChartDataS
  * @param state RechartsRootState
  * @returns data defined on the chart graphical items, such as Line or Scatter or Pie
  */
-export const selectAllGraphicalItemsData: (state: RechartsRootState) => ReadonlyArray<ChartData> = (
+export const selectCartesianGraphicalItemsData: (
   state: RechartsRootState,
-) => state.graphicalItems.graphicalItemData;
+  axisType: AxisType,
+  axisId: AxisId,
+) => ReadonlyArray<ChartData> = createSelector(
+  (state: RechartsRootState) => state.graphicalItems.cartesianItems,
+  (_state, axisType: AxisType) => axisType,
+  (_state, _axisType, axisId: AxisId) => axisId,
+  (cartesianItems: ReadonlyArray<CartesianGraphicalItemSettings>, axisType: AxisType, axisId: AxisId) =>
+    cartesianItems
+      .filter(item => {
+        if (axisType === 'xAxis') {
```

Comment:
what about y axes? or if the chart is vertical?

Reply:
To-do